### DISCUSSION TITLE
Does MOOSE support electrical simulation?

### DISCUSSION THREAD

**Discussion by tongfen** [idaholab/moose#17389] [Q&A Modules: General] [answered]
Hi, I am new to MOOSE. I am looking to simulate electrical-mechanical coupling. Does MOOSE support this? Thank you very much.

**Comment by cticenhour** [comment, marked as answer]
If all you need is an electrostatic field simulation, then the `Diffusion` kernel (or one of the custom versions: `FuncDiffusion` or `MatDiffusion` if you want to couple in a material property for the permittivity) could help you set up Poisson's Equation to solve for potential. Then you can set up an AuxKernel to get electric field as an AuxVariable from the electrostatic potential. If you need full Maxwell's, an electromagnetic module for MOOSE has been built but is not cleared for public release yet. It should be coming very soon though.

**Reply by tongfen** [reply]
Hi Cticenhour, 
Thank you so much for your information. It is still beyond my understanding. I will spend some time to figure it out. Thanks again!

**Reply by cticenhour** [reply]
No problem! Feel free to post more questions you begin learning MOOSE.

**Comment by mangerij** [comment]
Are you asking about piezoelectric materials?

**Reply by tongfen** [reply]
Not exactly, do you have any idea about using MOOSE to simulate piezoresistive materials? Thanks.

**Reply by mangerij** [reply]
In my limited knowledge of piezoresistivity, wouldn't you just need to couple elastic deformations to a change in conductivity? This would just be a postprocessor added to what is already in Moose: tensor mechanics

Or is there coupling between the current and the strain?

**Reply by dschwen** [reply]
> This would just be a postprocessor

More likely a Material that couples in the stress or strain and computes a Diffusivity (or electrical conductivity) at each quadrature point.

**Reply by tongfen** [reply]
Mangerij and Dschwen, I see. Thank you for your reply. I got some useful information from you both.

**Comment by dschwen** [comment]
See also https://github.com/idaholab/moose/issues/17362
